test/test.js

```diff
@@ -5,111 +5,598 @@ suite('GaiaHeader', function() {
+    var el = this.dom.firstElementChild;
+    var actionButton = el.shadowRoot.querySelector('.action-button');
+
+    el.addEventListener('action', () => done());
+    actionButton.click();
   });
 
   test('triggerAction() should cause a `click` on action button', function() {
-    this.clock = sinon.useFakeTimers();
-    this.container.innerHTML = '<gaia-header action="menu"></gaia-header>';
-    var element = this.container.firstElementChild;
+    this.sinon.useFakeTimers();
+
+    this.dom.innerHTML = '<gaia-header action="menu"></gaia-header>';
+    var el = this.dom.firstElementChild;
     var callback = sinon.spy();
-    element.addEventListener('action', callback);
-    element.triggerAction();
-    this.clock.tick(1);
+
+    el.addEventListener('action', callback);
+    el.triggerAction();
+    this.sinon.clock.tick();
     assert.equal(callback.args[0][0].detail.type, 'menu');
-    this.clock.restore();
   });
 
-  test('It fails silently when `window.getComputedStyle()` returns null (ie. hidden iframe)', function() {
-    this.sandbox.stub(window, 'getComputedStyle').returns(null);
-    this.container.innerHTML = '<gaia-header action="menu"><h1>title</h1></gaia-header>';
-    var element = this.container.firstElementChild;
+  test('runFontFit does not b0rk the markup', function(done) {
+    this.dom.innerHTML = '<gaia-header action="back"><h1><p>markup</p></gaia-header>';
+    var el = this.dom.firstElementChild;
+
+    afterNext(el, 'runFontFit').then(() => {
+      assert.isNotNull(el.querySelector('p'));
+    }).then(done, done);
+  });
+
+  test('It still works fine after detaching and reattaching', function(done) {
+    this.dom.innerHTML = '<gaia-header action="menu"><h1>title</h1></gaia-header>';
+    var el = this.dom.firstElementChild;
+    var h1 = el.querySelector('h1');
+
+    this.sinon.spy(el, 'observerStop');
+    this.sinon.spy(el, 'observerStart');
+
+    afterNext(el, 'runFontFit').then(() => {
+      el.runFontFit.reset();
+
+      // Detach from DOM
+      el.remove();
+
+      sinon.assert.called(el.observerStop, 'observer disconnected');
+
+      // Change the title text when
+      // the element is detached
+      h1.textContent = 'bar';
+
+      return afterNext(el, 'runFontFit').catch(() => {
+
+        sinon.assert.notCalled(this.runFontFit, 'font-fit doesn\'t run when detached');
+
+        // Reattach the element
+        this.dom.appendChild(el);
+
+        return afterNext(el, 'runFontFit').then(() => {
+          sinon.assert.called(el.runFontFit, 'font-fit run when re-attached');
+          sinon.assert.called(el.observerStart, 'observer started again');
+
+          h1.textContent = 'foo';
 
-    // Insert into DOM to get styling
-    document.body.appendChild(element);
+          return afterNext(el, 'runFontFit').then(() => {
+            sinon.assert.called(el.runFontFit, 'font-fit run on mutation');
+          });
+        });
+      });
```

Comment:
same general comments about the indentation :) you can return a promise from the `then` handler, and its status is propagated to the promise returned by the `then` method.